Xamarin.Forms.Core/Shell/Shell.cs

```diff
@@ -314,7 +317,24 @@ async Task IShellController.OnFlyoutItemSelectedAsync(Element element)
 			if (FlyoutIsPresented && FlyoutBehavior == FlyoutBehavior.Flyout)
 				SetValueFromRenderer(FlyoutIsPresentedProperty, false);
 
-			await GoToAsync(state).ConfigureAwait(false);
+			if (shellSection == null)
+				shellItem.PropertyChanged += OnShellItemPropertyChanged;
+			else if (shellContent == null)
+				shellSection.PropertyChanged += OnShellItemPropertyChanged;
+			else
+				await GoToAsync(state).ConfigureAwait(false);
+		}
+
+		void OnShellItemPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (e.PropertyName == CurrentItemProperty.PropertyName)
+			{
+				(sender as BindableObject).PropertyChanged -= OnShellItemPropertyChanged;
```

Comment:
Why do we unsubscribe here? Not understanding very well, can we go back and if i click again since we unsubscribed it will not navigate again

Reply:
The only point of this code is to delay navigation until a visible stack of Shell Elements are realized so it's only relevant as the navigation is happening and the visible items are showing up.

The code gets here
https://github.com/xamarin/Xamarin.Forms/pull/7847/files/528247083f733948659463b1a2b9ea6e3089f647#diff-070ffbecea113674cafb32e2f6776e08R321

It sees that CurrentItem is null so I subscribe to an event until there's a valid ShellItem.ShellSection.ShellContent

Then it navigates there. So this is only relevant as it's navigating not to initiate navigation